File: app/src/test/java/com/example/benchmarks/OperationsMapsUnitTest.java
package com.example.benchmarks;

import static org.junit.Assert.*;

import com.example.benchmarks.models.Benchmark;
import com.example.benchmarks.models.BenchmarkItem;
import com.example.benchmarks.models.OperationsMaps;

import org.junit.Test;

import java.util.List;


public class OperationsMapsUnitTest {

    Benchmark benchmark = new OperationsMaps();
    List<BenchmarkItem> testList = benchmark.createBenchmarkList(false);

    @Test
    public void testCreatingBenchmarkListReturnsNotNull() {
        assertNotNull(testList);
    }

    @Test
    public void testCreatingBenchmarkListIsNotEmpty(){
        assertTrue(testList.size() > 0);
    }

    @Test
    public void testMeasureTimeMoreThan0(){
        for (BenchmarkItem item : testList) {
            assertTrue(benchmark.measureTime(5000, item) > 0);
        }
    }

    @Test
    public void testMeasureTimeThrowsException(){
        BenchmarkItem wrongItem = new BenchmarkItem(1, 1, 0L, false);
        assertThrows(RuntimeException.class, () -> benchmark.measureTime(10, wrongItem));
    }

    @Test
    public void testGetSpansCount(){
        assertEquals(2, benchmark.getSpansCount());
    }

}

File: app/src/test/java/com/example/benchmarks/models/benchmark/OperationsMapsUnitTest.kt
package com.example.benchmarks.models.benchmark

import org.junit.Assert.*
import org.junit.Test

class OperationsMapsUnitTest {

    private val benchmark: Benchmark = OperationsMaps()

    @Test
    fun testCreatingBenchmarkListReturnsNotNull() {
        val testList = benchmark.createBenchmarkList(false)
        assertNotNull(testList)
    }

    @Test
    fun testCreatingBenchmarkListIsNotEmpty() {
        val testList = benchmark.createBenchmarkList(false)
        assertTrue(testList.isNotEmpty())
    }

    @Test
    fun testMeasureTimeMoreThan0() {
        val testList = benchmark.createBenchmarkList(false)
        for (item in testList) {
            assertTrue(benchmark.measureTime(5000, item) > 0)
        }
    }

    @Test
    fun testMeasureTimeThrowsException() {
        val wrongItem = BenchmarkItem(1, 1, 0L, false)
        assertThrows(RuntimeException::class.java) { benchmark.measureTime(10, wrongItem) }
    }

    @Test
    fun testGetSpansCount() {
        assertEquals(2, benchmark.getSpansCount())
    }

}